clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -196,6 +197,44 @@ bool Sema::isKnownGoodSYCLDecl(const Decl *D) {
   return false;
 }
 
+bool isArraySizedZero(QualType Ty) {
+  if (const auto *CATy = dyn_cast<ConstantArrayType>(Ty)) {
+    const llvm::APInt size = CATy->getSize();
+    return size == 0;
+  }
+  return false;
+}
+
+void Sema::CheckVarDeclOKIfInKernel(VarDecl *var) {
```

Comment:
Variable names start with a capital letter by coding standard: https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly